java/core/src/java/org/apache/orc/impl/ReaderImpl.java

```diff
@@ -341,7 +338,9 @@ public ReaderImpl(Path path, OrcFile.ReaderOptions options) throws IOException {
     } else {
       OrcTail orcTail = options.getOrcTail();
       if (orcTail == null) {
-        tail = extractFileTail(fs, path, options.getMaxLength());
+        this.fileSystem = getFileSystem();
+        options.filesystem(fileSystem);
```

Comment:
options fs object got lost - it's not clear that was null before.

Reply:
fixed it.